### DISCUSSION TITLE
`poetry update pkg_a` not update anything

### DISCUSSION THREAD

**Discussion by Albert-Gao** [python-poetry/poetry#8245] [General] [answered]
Cost 10min to figure out how to update a package.

basically, i found the outdated packages by running this.
```bash
poetry show --latest --outdated
```

then i use this
```bash
poetry update pkg_a
```

say `pkg_a` is an outdated package, but the runtime shows `No dependencies to install or update`

Looks like a bug,

the only way to fix this bug is to manually update the version in the `pyproject.toml`

what am i missing here? thanks. :)

**Comment by radoering** [comment, marked as answer]
`poetry update` only updates the lockfile according to the constraints in the `pyproject.toml`. Try `poetry add pkg_a@latest`.

**Reply by Albert-Gao** [reply]
Thanks, now I understand. it feels a bit confusing TBH 😅
